Communication guidelines

marijnschilling opened the pull request:
Oke, here is the first draft of some guidelines for our communication. Im curious to your thoughts about these, I think these small adjustments might make our communication way more efficient.

Ha there's probably a lot of typing and grammar mistakes in it, wrote it quite quickly before lunch!

Anddd:  keep the guidelines in mind when replying! 😬✌

marijnschilling committed:
Add guidelines for communication

marijnschilling committed:
Apply some styling

3lvis commented:
Great initiative, @marijnschilling! 👏

Elland commented:
👏 I hope this helps us! Great idea.

Do you want some spelling/grammar review? 🤔  What about formatting? I think using blockquotes for some of the stuff there looked a bit odd.

marijnschilling committed:
Add advice to solve the issue via a call

marijnschilling commented:
Where am i using blockquotes exactly? I don't really know how I can see how it would look on GitHub eventually. I use this markdown editor: http://dillinger.io/

It would be great if you could correct spelling, grammar and formatting in a PR. Than I could continue working on other things. :V:

Elland commented:
How to preview how it will look on GitHub:
<img width="1351" alt="screen shot 2016-11-23 at 13 03 49" src="https://cloud.githubusercontent.com/assets/54423/20561171/710738ec-b17d-11e6-8ca3-fa23e1bb12c7.png">

I never head of Dillinger, but it does live preview. 🤔 But since it has its own stylesheet it looks quite different from how GitHub does it. It's worth checking out how it will look on GitHub. ✨ 

I'll go over the text shortly and open one with spelling, grammar, formatting suggestions. 👍

marijnschilling commented:
Ah great! I see the button now 👯

Elland commented:
💥 #76

elland committed:
Communication guidelines draft #2. (#76)

* Communication guidelines draft #2.

- I first went over spelling/grammar, formatting.
- I also found that some points could be better clarified, some examples could be improved and some parts could use a different tone.
- I hope this reads nicely. 🎉

* Update COMMUNICATION.md

* Update COMMUNICATION.md

* Update COMMUNICATION.md

* Update COMMUNICATION.md

* Update COMMUNICATION.md

Elland commented:
👏

marijnschilling committed:
Update COMMUNICATION.md

Clarify the reason why explaining what you are replying to is important

marijnschilling commented:
Forgot one tiny thing in your PR @elland, updated in last commit. Ready to merge? 👏

3lvis commented:
Weird that it doesn't link. Damn, GitHub. 

@marijnschilling  -> https://github.com/bakkenbaeck/iOS-playbook/pull/78

elland commented on COMMUNICATION.md:
```diff
@@ -15,7 +15,7 @@ When you write a reply of less than ~5 words, have an alarm bell in the back of
 
 ## Criticising or asking questions
-Start by repeating **what** you are replying to in your own words. This avoids cases where due to the high volume of messages, the context of your comment is lost. It also helps you make sure you yourself fully understood the problem, and didn’t misread anything.
+Start by repeating **what** you are replying to in your own words. This forces you to indulge yourself into the perspective of the other, which creates a greater understanding of the issue both for yourself and the other when reading your reply.
```
I think we agreed to leave this one as is, as this is just repeating the bit about empathy we already have, instead of providing more context? 🤔

elland commented on COMMUNICATION.md:
```diff
@@ -1,7 +1,7 @@
 
 ## Why we have guidelines about communicating
-Within our iOS team we mostly communicate with each other via the magic of the internet, and that is great! It's easy and very efficient. But online communication can easily result into misunderstandings, which tend to be long, time consuming and inefficient. That's why we created some guidelines for how we like to communicate professionally, keep them in the back of your head when commenting on  ...
+Within our iOS team we mostly communicate with each other via the magic of the internet, and that is great! It's easy and very efficient. But online communication can easily result into misunderstandings, which tend to be long, time consuming and inefficient. That's why we created some guidelines for how we like to communicate professionally, keep them in the back of your head when commenting on  ...
```
🙈

elland commented on COMMUNICATION.md:
```diff
@@ -40,7 +40,7 @@ Explain why you think this might be helpful in this particular case. Don’t jus
 If you feel someone’s reaction is not clear enough (i.e. someone’s reaction confuses you, or even agitates you), look at these guidelines together to see what could be improved in the future. If you realise something critical is missing in the guidelines, please add to it.
 
-Try to keep in mind that written communication it’s really easy to misread something or to interpret something out of context. Try not to assume rudeness, ill intent or malice. Sometimes something was poorly phrased os misread. Sometimes we’re switching between contexts too often, and end up being shorter than we should. Try asking for clarification, or if the person could rephrase that, if the m ...
+Try to keep in mind that written communication it’s really easy to misread something or to interpret something out of context. Try not to assume rudeness, ill intent or malice. Sometimes something was poorly phrased or misread. Sometimes we’re switching between contexts too often, and end up being shorter than we should. Try asking for clarification, or if the person could rephrase that, if the m ...
```
🙈 ✨

elland commented on COMMUNICATION.md:
```diff
@@ -0,0 +1,63 @@
+- Be humble. (“I’m not sure, let’s try…”)
+- Never deal in absolutes. Unless needed. (“We should never store sensitive data in plain text.”)
+- Aim to develop professional skills, group knowledge and product quality, through group critique. That goes both way: try to be open and welcoming to criticism.
+- Be aware of negative bias with online communication. (If content is neutral, we assume the tone is negative.) Try to use positive language instead.
```
"Try using positive"

3lvis replied:
This fixe should go in https://github.com/bakkenbaeck/iOS-playbook/blob/master/GIT_AND_GITHUB.md

elland commented on COMMUNICATION.md:
```diff
@@ -0,0 +1,63 @@
+- Avoid using derogatory terms, like “stupid”, when referring to the work someone has produced. Don’t shame, no one started out knowing.
+- Be humble. (“I’m not sure, let’s try…”)
+- Never deal in absolutes. Unless needed. (“We should never store sensitive data in plain text.”)
+- Aim to develop professional skills, group knowledge and product quality, through group critique. That goes both way: try to be open and welcoming to criticism.
```
"That goes both way**s**:"

3lvis replied:
This fixe should go in https://github.com/bakkenbaeck/iOS-playbook/blob/master/GIT_AND_GITHUB.md

elland replied:
@3lvis well I did point them there as well. 👌 But the issue is now merged?

3lvis replied:
@Elland I didn't see your comments. You can always fix it yourself now.

elland reviewed (commented):
Some of my spelling errors passed unnoticed.

Elland commented:
I think that was it. Just the spelling here and there and we can merge this! 💞 

Great work everyone.

3lvis commented:
I would like to do a final read after https://github.com/bakkenbaeck/iOS-playbook/pull/78 is merged. :)

Elland commented:
Why after?

3lvis commented:
@Elland Because that's how it would look.

Elland commented:
I don't understand. 😕 ❓ Why not read it now in Preview mode?

3lvis commented:
@Elland doesn't make much difference really. :) I can do it now, just for you.

3lvis commented on README.md:
```diff
@@ -3,6 +3,7 @@
 Consistency is 👸.
 
 ### Index
+* [Communication](/COMMUNICATION.md)
```
💥

Elland commented:
@3lvis just that by doing it now, if we catch more errors, we can fix them before publishing. 😄

3lvis commented on COMMUNICATION.md:
```diff
@@ -0,0 +1,63 @@
+- **When writing short replies**
+When you write a reply of less than ~5 words, have an alarm bell in the back of your head 🔔. Is it just tacit agreement? Then it’s probably fine! “Great job”, “Couldn’t have put it better”,  “toats awesome!” are all fine. In every other case though: 🚨!
+
+It good practice to elaborate. It may seem inefficient to be so verbose in our replies, but it will avoid misunderstandings, which can result in way greater inefficiency. Avoid being pennywise, pound foolish.
```
>Avoid being pennywise, pound foolish.

I think we should drop this or put it in other terms, what's the intent?

elland replied:
To avoid being pennywise, pound foolish? 🙈 🙊 😳

3lvis replied:
That's probably redundant to what came before, right?

marijnschilling replied:
Good point, let's remove it! it's a english proverb, but it only repeats what has been said before. If misunderstood it can only lead to confusion.

elland replied:
More like expanding on it a bit? Just something to avoid in general, not just in the measures of our replies?

marijnschilling replied:
Using proverbs can be tricky, cause there less widely known and thus can lead to confusion.

3lvis replied:
🔪 kill it!

elland replied:
More likely to lead to learning, in my experience. It's also not really redundant, as it expands the suggestion to ask you to take that mindset into consideration not only for communication but for other aspects as well. It's also not a proverb, more like a phrasal adjective. 🤔

marijnschilling committed:
Update COMMUNICATION.md

3lvis commented on COMMUNICATION.md:
```diff
@@ -0,0 +1,63 @@
+- What’s your suggestion?
+
+### Preferred:
+- **I see you are using an optional as an array of User object in the example above**, wouldn’t it be better to have that property default to an empty array?
```
Instead of:
>wouldn’t it be better to have that property default to an empty array?

I would go for:
>have you considered to have that property default to an empty array?

There's a chance that it was already considered and you might end up learning something on why it wasn't done that way. Meanwhile on the first one you are suggesting they didn't know something or tried something in advance without asking.

marijnschilling committed:
Update COMMUNICATION.md

3lvis commented on COMMUNICATION.md:
```diff
@@ -0,0 +1,63 @@
+## When you feel someone is not communicating clear enough
+If you feel someone’s reaction is not clear enough (i.e. someone’s reaction confuses you, or even agitates you), look at these guidelines together to see what could be improved in the future. If you realise something critical is missing in the guidelines, please add to it.
+
+Try to keep in mind that written communication it’s really easy to misread something or to interpret something out of context. Try not to assume rudeness, ill intent or malice. Sometimes something was poorly phrased or misread. Sometimes we’re switching between contexts too often, and end up being shorter than we should. Try asking for clarification, or if the person could rephrase that, if the m ...
```
>if the meaning is’t clear 

There's a typo there 😁  . Should be:

>if the meaning isn’t clear

3lvis commented:
This is really good, @marijnschilling.

![giphy 4](https://cloud.githubusercontent.com/assets/1088217/20596489/1c8ecc54-b23f-11e6-88ad-69158e9e80ec.gif)

marijnschilling committed:
Update COMMUNICATION.md
